voice/voice-ai/x/AppSettings/SettingsView.swift

```diff
@@ -77,4 +95,37 @@ struct SettingsView: View {
         }
         print("Settings: system settings clicked")
     }
+    
+    func getUserName() -> String {
+        let keychain = KeychainService.shared
+        if keychain.isAppleIdAvailable() {
+            guard let email = keychain.retrieveEmail(), email.isEmpty else {
+                let userID =  keychain.retrieveUserid() ?? "User id not found"
+                return userID
+            }
+            return email
+        }
+        return "Sign In"
+    }
+    
+    func saveTranscript() {
+        if SpeechRecognition.shared.conversation.isEmpty {
+            showAlert = true
+            return
+        }       
+        isSaveTranscript = true
+    }
+  
+    func convertMessagesToTranscript(messages: [Message]) -> String {
+        var transcript = ""
+
+        for message in messages {
+            let label = message.role?.lowercased() == "user" ? "User:" : "GPT:"
```

Comment:
Might need to use `AI` since `GPT` is trademarked. We might also need to switch to other models anyway